src/lib/bookmarks/bookmarkstoolbar.cpp

```diff
@@ -72,10 +73,14 @@ void BookmarksToolbar::contextMenuRequested(const QPoint &pos)
     menu.addSeparator();
     QAction* actDelete = menu.addAction(QIcon::fromTheme("edit-delete"), tr("Delete"));
     menu.addSeparator();
-    QAction* act = menu.addAction(tr("Show Only Icons"));
-    act->setCheckable(true);
-    act->setChecked(m_bookmarks->showOnlyIconsInToolbar());
-    connect(act, SIGNAL(toggled(bool)), m_bookmarks, SLOT(setShowOnlyIconsInToolbar(bool)));
+    m_actShowOnlyIcons = menu.addAction(tr("Show Only Icons"));
```

Comment:
Last thing, either create the actions in constructor, or null them after closing the menu (as with `m_clickedBookmark`).